Calibration/HcalCalibAlgos/src/AnalyzerMinbias.cc

```diff
@@ -0,0 +1,537 @@
+    DetId id = (*hbheItr).detid(); 
+    HcalDetId hid=HcalDetId(id);
+    
+    std::map<std::pair<int,HcalDetId>,myInfo>::iterator itr1 = myMap_.find(std::pair<int,HcalDetId>(algoBit,hid));
+    
+    if (itr1 == myMap_.end()) {
+      myInfo info;
+      myMap_[std::pair<int,HcalDetId>(algoBit,hid)] = info;
+      itr1 = myMap_.find(std::pair<int,HcalDetId>(algoBit,hid));
+    } 
+    itr1->second.theNS0++;
+    itr1->second.theNS1 += energyhit;
+    itr1->second.theNS2 += (energyhit*energyhit);
+    itr1->second.theNS3 += (energyhit*energyhit*energyhit);
+    itr1->second.theNS4 += (energyhit*energyhit*energyhit*energyhit);
+    itr1->second.runcheck = rnnum;
+    if (fill) h_Noise[hid.subdet()-1]->Fill(energyhit);
+ 
+    std::map<std::pair<int,HcalDetId>,myInfo>::iterator itr2 = tmpMap.find(std::pair<int,HcalDetId>(algoBit,hid));
+    if (itr2 == tmpMap.end()) {
+      myInfo info;
+      tmpMap[std::pair<int,HcalDetId>(algoBit,hid)] = info;
+      itr2 = tmpMap.find(std::pair<int,HcalDetId>(algoBit,hid));
+    }
+    itr2->second.theNS0++;
+    itr2->second.theNS1 += energyhit;
+    itr2->second.theNS2 += (energyhit*energyhit);
+    itr2->second.theNS3 += (energyhit*energyhit*energyhit);
+    itr2->second.theNS4 += (energyhit*energyhit*energyhit*energyhit);
+    itr2->second.runcheck = rnnum;
+    
+  } // HF_NS
+	  
+	  
+    // Signal part for HF
+	  
+  for (HFRecHitCollection::const_iterator hbheItr=HithfMB.begin(); 
+       hbheItr!=HithfMB.end(); hbheItr++) {
+    // Recalibration of energy
+    float icalconst=1.;	 
+    DetId mydetid = hbheItr->id().rawId();
+    if (theRecalib_) icalconst=myRecalib->getValues(mydetid)->getValue();
+    HFRecHit aHit(hbheItr->id(),hbheItr->energy()*icalconst,hbheItr->time());
+    
+    double energyhit = aHit.energy();
+    // Remove PMT hits
+    if(fabs(energyhit) > 40. ) continue;
+	      
+    DetId id = (*hbheItr).detid(); 
+    HcalDetId hid=HcalDetId(id);
+	      
+    std::map<std::pair<int,HcalDetId>,myInfo>::iterator itr1 = myMap_.find(std::pair<int,HcalDetId>(algoBit,hid));
+    std::map<std::pair<int,HcalDetId>,myInfo>::iterator itr2 = tmpMap.find(std::pair<int,HcalDetId>(algoBit,hid));
+    
+    if (itr1 == myMap_.end()) {
+      myInfo info;
+      myMap_[std::pair<int,HcalDetId>(algoBit,hid)] = info;
+      itr1 = myMap_.find(std::pair<int,HcalDetId>(algoBit,hid));
+    }
+    itr1->second.theMB0++;
+    itr1->second.theDif0 = 0;
+    itr1->second.theMB1 += energyhit;
+    itr1->second.theMB2 += (energyhit*energyhit);
+    itr1->second.theMB3 += (energyhit*energyhit*energyhit);
+    itr1->second.theMB4 += (energyhit*energyhit*energyhit*energyhit);
+    itr1->second.runcheck = rnnum;
+    float mydiff = 0.0;
+    if (itr2 !=tmpMap.end()) {
+      mydiff = energyhit - (itr2->second.theNS1);
+      itr1->second.theDif0++;
+      itr1->second.theDif1 += mydiff;
+      itr1->second.theDif2 += (mydiff*mydiff);
+      if (fill) h_Signal[hid.subdet()-1]->Fill(mydiff);
```

Comment:
One point on using TFileService. One either has to use a 'legacy' module type or inherit from ```edm::one::EDProducer<edm::one::SharedResources>``` and in the constructor have 
```
usesResource(edm::TFileService::kSharedResource);
```